sklearn/metrics/tests/test_common.py

```diff
@@ -1110,3 +1110,101 @@ def test_no_averaging_labels():
+    y_true_perm = classes_perm[y_true]
+    y_pred_perm = classes_perm[y_pred]
+    for name in CLASSIFICATION_METRICS:
+        if name in METRIC_UNDEFINED_BINARY_MULTICLASS:
+            continue
+        if name == "confusion_matrix":
+            continue
+        metric = ALL_METRICS[name]
+        score = metric(y_true, y_pred)
+        score_perm = metric(y_true_perm, y_pred_perm)
+        assert_almost_equal(score, score_perm)
+
+
+def test_metric_permutation_invariance_multilabel():
+    rng = np.random.RandomState(42)
+    n_samples = 100
+    n_classes = 5
+    p = 0.7
+    R = np.random.rand(n_samples, n_classes)
+    y_true = R > p
+    y_pred = R + np.random.randn(n_samples, n_classes) > p
+    classes_perm = rng.permutation(n_classes)
+    y_true_perm = y_true[:, classes_perm]
+    y_pred_perm = y_pred[:, classes_perm]
+
+    for name in MULTILABELS_METRICS:
+        metric = ALL_METRICS[name]
+        score = metric(y_true, y_pred)
+        score_perm = metric(y_true_perm, y_pred_perm)
+        assert_almost_equal(score, score_perm)
+
+
+def test_metric_permutation_invariance_thresholded_multiclass():
+    rng = np.random.RandomState(42)
+    y_score = rng.rand(100, 3)
+    y_score[:, 2] += .1
+    y_score[:, 1] -= .1
+    y_true = np.argmax(y_score, axis=1)
+    y_true[rng.randint(len(y_score), size=20)] = np.random.randint(
+        2, size=20)
+    for name in THRESHOLDED_METRICS:
+        if name in METRIC_UNDEFINED_BINARY_MULTICLASS:
+            continue
+        metric = ALL_METRICS[name]
+        same_score_under_permutation = None
+        for perm in [[0, 1, 2], [0, 2, 1], [1, 0, 2],
+                     [1, 2, 0], [2, 0, 1], [2, 1, 0]]:
+            inv_perm = np.zeros(3, dtype=int)
+            inv_perm[perm] = np.arange(3)
+            y_score_perm = y_score[:, inv_perm]
+            y_true_perm = np.take(perm, y_true)
+            score = metric(y_true_perm, y_score_perm)
+            if same_score_under_permutation is None:
+                same_score_under_permutation = score
+            else:
+                assert_almost_equal(score, same_score_under_permutation)
+
+
+def test_metric_permutation_invariance_thresholded_multilabel():
+    rng = np.random.RandomState(42)
+    n_samples = 100
+    n_classes = 5
+    p = 0.7
+    R = np.random.rand(n_samples, n_classes)
+    y_true = R > p
+    y_score = []
+    y_score_perm = []
+    classes_perm = rng.permutation(n_classes)
+    for i in range(len(y_true)):
+        while len(np.unique(y_true[i])) == 1:
+            y_true[i] = np.random.rand(n_classes) > p
+        y_score.append(y_true[i] + rng.normal(size=y_true[i].shape))
+        y_score_perm.append(y_score[i][classes_perm])
+
+    y_true_perm = y_true[:, classes_perm]
+    for name in THRESHOLDED_MULTILABEL_METRICS:
+        metric = ALL_METRICS[name]
+        score = metric(y_true, y_score)
+        score_perm = metric(y_true_perm, y_score_perm)
+        assert_almost_equal(score, score_perm)
```

Comment:
Nice work here ladies, you all seem to have really catch up fast 👍 
The PR will be good to go when the other comments have been re-accessed. Thanks